react-frontend/Dockerfile-frontend

```diff
@@ -0,0 +1,45 @@
+# Build Stage
+FROM node:18 as build-stage
+
+WORKDIR /app
+
+# Copy package.json and package-lock.json to the working directory
+COPY package*.json ./
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application files to the working directory
+COPY . .
+
+# Build the React application for production
+RUN npm run build
```

Comment:
We build our project outside of docker, can we just copy our artifacts there? We do not need node modules and even nodejs here.

Reply:
And then just build the image in github workflow

Reply:
I am going to use docker hub for deployment and versioning, so I need seperate build process which is going to be triggered on every push to main branch. On pr I would use your CI to satisfy the requirements of the stage 2 and maybe in future I am going to add tests on backend and frontend. Moreover, I use docker while developing so I can't copy artifacts from github action.

Reply:
👍 makes sense